src/DynamoCoreWpf/ViewModels/Core/PortViewModel.cs

```diff
@@ -412,6 +417,49 @@ private void OnMouseLeftUseLevel(object parameter)
             ShowUseLevelMenu = false;
         }
 
-       
+        /// <summary>
+        /// Returns a collection of node search elements for nodes
+        /// that output a type compatible with the port type if it's an input port.
+        /// These search elements can belong to either zero touch, NodeModel or Builtin nodes.
+        /// This method returns an empty collection if the input port type cannot be inferred or
+        /// there are no matching nodes found for the type. Currently the match is an exact match
+        /// done including the rank information in the type, e.g. Point[] or var[]..[].
+        /// The search elements can be made to appear in the node autocomplete search dialog.
+        /// </summary>
+        /// <returns>collection of node search elements</returns>
+        internal IEnumerable<NodeSearchElement> GetMatchingNodes()
```

Comment:
It seems we can move this into `NodeAutoCompleteSearchViewModel` so everytime we show the control we update the list of entries since I have the target PortViewModel stored as a property already. What do you think？

Reply:
Sure, we can do that. Will move this method.

Reply:
Also I have been thinking about maybe rename this method to `GetMatchingNodeSearchElements`? Since the API is not returning a collection of NodeModels. What do you think?